phpBB/includes/acp/acp_storage.php

```diff
@@ -96,8 +122,173 @@ public function overview($id, $mode)
+			{
+				$action = '';
+				$this->state = false;
+				$this->save_state();
+			}
+		}
+
+		if ($action)
+		{
+			switch ($action)
+			{
+				case 'progress_bar':
+					$this->display_progress_bar();
+				break;
+				case 'update':
+					// Just continue
+				break;
+				default:
+					trigger_error('NO_ACTION', E_USER_ERROR);
+				break;
+			}
+
+			if (!check_link_hash($this->request->variable('hash', ''), 'acp_storage'))
+			{
+				trigger_error($this->lang->lang('FORM_INVALID') . adm_back_link($this->u_action), E_USER_WARNING);
+			}
+
+			// If update_type is copy or move, copy files from the old to the new storage
+			if ($this->state['update_type'] >= 1)
+			{
+				$i = 0;
+				foreach ($this->state['storages'] as $storage_name => $storage_options)
+				{
+					// Skip storages that have already moved files
+					if ($this->state['storage_index'] > $i)
+					{
+						$i++;
+						continue;
+					}
+
+					$current_adapter = $this->get_current_adapter($storage_name);
+					$new_adapter = $this->get_new_adapter($storage_name);
+
+					$sql = 'SELECT file_id, file_path
+						FROM ' . STORAGE_TABLE . "
+						WHERE  storage = '" . $this->db->sql_escape($storage_name) . "'
+							AND file_id > " . $this->state['file_index'];
+					$result = $this->db->sql_query($sql);
+
+					$starttime = microtime(true);
+					while ($row = $this->db->sql_fetchrow($result))
+					{
+						if (!still_on_time())
+						{
+							$this->save_state();
+							meta_refresh(1, append_sid($this->u_action . '&amp;action=update&amp;hash=' . generate_link_hash('acp_storage')));
+							trigger_error($this->lang->lang('STORAGE_UPDATE_REDIRECT', $this->lang->lang('STORAGE_' . strtoupper($storage_name) . '_TITLE'), $i + 1, count($this->state['storages'])));
+						}
+
+						$stream = $current_adapter->read_stream($row['file_path']);
+						$new_adapter->write_stream($row['file_path'], $stream);
+
+						if (is_resource($stream))
+						{
+							fclose($stream);
+						}
+
+						$this->state['file_index'] = $row['file_id']; // Set last uploaded file
+					}
+
+					// Copied all files of a storage, increase storage index and reset file index
+					$this->state['storage_index']++;
+					$this->state['file_index'] = 0;
+				}
+
+				// If update_type is move files, remove the old files
+				if ($this->state['update_type'] == 2)
+				{
+					$i = 0;
+					foreach ($this->state['storages'] as $storage_name => $storage_options)
+					{
+						// Skip storages that have already moved files
+						if ($this->state['remove_storage_index'] > $i)
+						{
+							$i++;
+							continue;
+						}
+
+						$current_adapter = $this->get_current_adapter($storage_name);
+
+						$sql = 'SELECT file_id, file_path
+							FROM ' . STORAGE_TABLE . "
+							WHERE  storage = '" . $this->db->sql_escape($storage_name) . "'
+								AND file_id > " . $this->state['file_index'];
+						$result = $this->db->sql_query($sql);
+
+						$starttime = microtime(true);
```

Comment:
Please remove this not needed variable.

Reply:
It was in acp_search, i see it is referenced here https://github.com/phpbb/phpbb/blob/master/phpBB/includes/functions.php#L259 as global, but maybe is not the same

Reply:
Yes, probably i can remove it, i thought it was used by still_on_time